cpp/merge_module/algorithm/merge.cpp

```diff
@@ -1,11 +1,102 @@
 #include "merge.hpp"
-
 #include <algorithm>
 #include <optional>
 #include <string_view>
 #include <unordered_map>
 #include "mgp.hpp"
 
+bool Merge::LabelsContained(const std::unordered_set<std::string_view> &labels, const mgp::Node &node) {
+  bool contained = false;
+  auto size = labels.size();  // this works if labels are unique, which they are
+  size_t counter = 0;
+
+  for (const auto label : node.Labels()) {
+    if (labels.find(label) != labels.end()) {
+      counter++;
+    }
+  }
+
+  if (counter == size) {
+    contained = true;
+  }
+
+  return contained;
+}
+
+bool Merge::IdentProp(const mgp::Map &identProp, const mgp::Node &node) {
+  bool match = true;
+  for (const auto &[key, value] : identProp) {
+    if (value != node.GetProperty(std::string(key))) {
+      match = false;
+    }
+  }
+  return match;
+}
+
+void Merge::Node(mgp_list *args, mgp_graph *memgraph_graph, mgp_result *result, mgp_memory *memory) {
+  mgp::MemoryDispatcherGuard guard{memory};
+  const auto arguments = mgp::List(args);
+  const auto record_factory = mgp::RecordFactory(result);
+  try {
+    auto graph = mgp::Graph(memgraph_graph);
+    auto labels = arguments[0].ValueList();
+    auto identProp = arguments[1].ValueMap();
+    auto createProp = arguments[2].ValueMap();
+    auto matchProp = arguments[3].ValueMap();
+    std::unordered_set<std::string_view> label_set;
+    std::unordered_map<std::string_view, mgp::Value> identProp_map;
+    std::unordered_map<std::string_view, mgp::Value> createProp_map;
+    std::unordered_map<std::string_view, mgp::Value> matchProp_map;
```

Comment:
```suggestion
    std::unordered_map<std::string_view, mgp::Value> ident_prop_map;
    std::unordered_map<std::string_view, mgp::Value> create_prop_map;
    std::unordered_map<std::string_view, mgp::Value> match_prop_map;
```
Be consistent with namings here